t/grpc_server_example/main.go

```diff
@@ -217,6 +223,21 @@ func (s *server) Run(ctx context.Context, in *pb.Request) (*pb.Response, error)
 	return &pb.Response{Body: in.User.Name + " " + in.Body}, nil
 }
 
+func gRPCAndHTTPFunc(grpcServer *grpc.Server) http.Handler {
+	return h2c.NewHandler(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		mux := http.NewServeMux()
+		mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+			w.Write([]byte("hello http"))
+		})
+
+		if r.ProtoMajor == 2 && strings.Contains(r.Header.Get("Content-Type"), "application/grpc") {
```

Comment:
Try the https://pkg.go.dev/github.com/brankas/netmux.

Reply:
It looks like this is a library for routing distributions? It doesn't have any documentation description. For this test case, I think it would be better to be simple and direct, and there is no performance pursuit.

Reply:
No, this is for protocol sniffing.

Reply:
Sorry, I sent a wrong repo, please ignore it.